Core/Barcelona/IDS.swift

```diff
@@ -5,279 +5,204 @@
-    )
-
-    let completion: ([String: IDSState]) -> () = { mappedStates in
-        // And then save them to the cache
-        lazy var now = Date()
-        for state in mappedStates {
-            BLIDSIDQueryCache.shared.cache(state.value, for: state.key, time: now)
+    /// Wraps
+    /// `IDSIDQueryController.refreshIDStatus(forDestinations:service:listenerID:queue:errorCompletionBlock:)`
+    /// as an `async` method.
+    /// - Parameters:
+    ///   - destinations: String from `IDSCopyIDFor[handle type]Address` methods.
+    ///   - serviceIDSIdentifier: ``IMServiceStyle/idsIdentifier`` of the ``IMServiceStyle``.
+    /// - Returns: A dictionary mapping IDs to ``IDSState``  raw values through `NSNumber`s.
+    @MainActor
+    private static func refreshIDStatus(
+        for destinations: [String],
+        with serviceIDSIdentifier: String
+    ) async -> [String: NSNumber] {
+        await withCheckedContinuation { continuation in
+            sharedController.refreshIDStatus(
+                forDestinations: destinations,
+                service: serviceIDSIdentifier,
+                listenerID: idsListenerID,
+                queue: handleQueue
+            ) {
+                continuation.resume(returning: $0)
+            }
         }
-
-        // And return them in the callback
-        log.debug("forceRefreshIDStatus completed with result: \(mappedStates)")
-        callback(mappedStates.mapKeys(\.idsURIStripped))
     }
 
-    let sharedController: IDSIDQueryController = IDSIDQueryController.sharedInstance()!
-
-    if #available(macOS 13.0, *) {
-        sharedController.idInfo(
-            forDestinations: destinations,
-            service: service.idsIdentifier,
-            // I don't quite know what `infoTypes` is asking for, but ChatKit passes in 0x2
-            infoTypes: 0x2,
-            options: .refreshIDInfo(),
-            listenerID: IDSListenerID,
-            queue: HandleQueue
-        ) {
-            callback($0.mapValues { IDSState(rawValue: $0.status()) })
-        }
-    } else {
-        sharedController.forceRefreshIDStatus(
-            forDestinations: destinations,
-            service: service.idsIdentifier,
-            listenerID: IDSListenerID,
-            queue: HandleQueue
-        ) { states in
-            completion(states.mapValues { IDSState.init(rawValue: $0.intValue) })
+    /// Wraps
+    /// `IDSIDQueryController.refreshIDStatus(forDestinations:service:listenerID:queue:errorCompletionBlock:)`
+    /// as an `async` method.
+    /// - Parameters:
+    ///   - destinations: String from `IDSCopyIDFor[handle type]Address` methods.
+    ///   - serviceIDSIdentifier: ``IMServiceStyle/idsIdentifier`` of the ``IMServiceStyle``.
+    /// - Returns: A dictionary mapping IDs to `IDSIDInfoResult`.
+    @available(macOS 13.0, *)
+    @MainActor
+    private static func idInfo(
```

Comment:
It seems that this function isn't referenced at all in this new code - was that an oversight?